### PR TITLE
Fix/70977 meta api

### PR DESCRIPTION
**Pull request by jesseeproductions**
_Ref:_ [C#70977](https://central.tri.be/issues/70977)

See -> https://github.com/moderntribe/event-tickets-plus/pull/314

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | empty() and isset() must only contain variables (PHP 5.2 compatibility) | [Tickets.php:1451](https://github.com/moderntribe/event-tickets/blob/3db2cc0a7989a83276478119108abfabf9ea99cd/src/Tribe/Tickets.php#L1451)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Tickets.php [hunk 1]
```diff
@@ -1417,6 +1417,56 @@ protected function get_attendee_event_key( $provider_class ) {
...
+					$value        = array();
+
+					foreach ( $meta as $full_key => $check_value ) {
+						if ( 0 === strpos( $full_key, $field_prefix ) ) {
+							$short_key           = substr( $full_key, strlen( $field_prefix ) );
+							$value[ $short_key ] = $check_value;
+						}
+					}
+
+					if ( empty ( $value ) ) {
```
Comment (bordoni) [inline src/Tribe/Tickets.php]: Space between `empty` and `(` shouldn't be there.

File: src/Tribe/Data_API.php [hunk 2]
```diff
@@ -66,18 +66,23 @@ protected function setup_data() {
...
 	 */
-	public function detect_by_id( $post_id ) {
+
+	public function detect_by_id( $post = null ) {
+
 		// only the rsvp order key is non numeric
-		if ( ! is_numeric( $post_id ) ) {
-			$post_id = esc_attr( $post_id );
-			$cpt     = $this->check_rsvp_order_key_exists( $post_id );
+		if ( ! is_numeric( $post ) ) {
```
Comment (barryhughes) [inline src/Tribe/Data_API.php]: Won't this also catch objects with an ID property (that would also not be numeric) ... should the clauses be re-ordered?

File: src/Tribe/Data_API.php [hunk 3]
```diff
@@ -66,18 +66,23 @@ protected function setup_data() {
 	/**
 	 * Detect what is available in the custom post type by the id passed to it
 	 *
-	 * @param $post_id
+	 * @param null $post id or post object
 	 *
 	 * @return array|bool array includes infomation available and the tribe_tickets_tickets class to use
 	 */
-	public function detect_by_id( $post_id ) {
+
```
Comment (barryhughes) [inline src/Tribe/Data_API.php]: This line break between docblock and method definition should be removed

**Pull request merged**